client/sections.js

```diff
@@ -61,6 +61,11 @@ sections = [
 		enableLoggedOut: true
 	},
 	{
+		name: 'start-trial',
+		paths: [ '/start-trial' ],
```

Comment:
I like the fact that we don't touch the `<Checkout />` component nor the upgrades controller whose refactoring would be time-consuming but I feel that `/checkout/trial` would be more appropriate as url. What were the reasons for opting for `start-trial`?